vsphere/resource_vsphere_host.go

```diff
@@ -817,3 +913,206 @@ func (h HostAccessManager) ChangeLockdownMode(ctx context.Context, mode types.Ho
+				}
+			}
+
+			// Update NTP servers
+			interfaces := ntpdConfig["ntp_servers"].([]interface{})
+			newServers := make([]string, len(interfaces))
+			for i, server := range interfaces {
+				newServers[i] = server.(string)
+			}
+
+			err := changeHostNtpServers(context.Background(), hostObject, newServers)
+			if err != nil {
+				return fmt.Errorf("error starting ntpd service on host %s: %v", hostID, err)
+			}
+			// Update the ntpdConfig map before setting it
+			updatedNtpdConfig["enabled"] = ntpdConfig["enabled"]
+			updatedNtpdConfig["ntp_servers"] = newServers // Updated servers
+			updatedNtpdConfig["policy"] = ntpdConfig["policy"]
+
+			// Update the NTP service policy if applicable
+			if policy, ok := ntpdConfig["policy"].(string); ok {
+				err := UpdateHostServicePolicy(context.Background(), client, hostObject, "ntpd", policy)
+				if err != nil {
+					return fmt.Errorf("error updating NTP servers for host %s. Error: %s", hostID, err)
+				}
+			}
+
+			updatedServiceMap["ntpd"] = []interface{}{updatedNtpdConfig}
+		}
+		// Handle other services similarly and add to updatedServiceMap as needed
+
+		updatedServices = append(updatedServices, updatedServiceMap) // Add the updated service map to the collection
+	}
+
+	// After processing all services, set the entire updated services configuration
+	if err := d.Set("services", updatedServices); err != nil {
+		return fmt.Errorf("error updating ntpd service policy on host %s: %v", hostID, err)
+	}
+
+	return nil
+}
+
+func changeHostNtpServers(ctx context.Context, host *object.HostSystem, servers []string) error {
+	s, err := host.ConfigManager().DateTimeSystem(ctx)
+	if err != nil {
+		// Handle the error, for example, by returning it
+		return fmt.Errorf("failed to get DateTimeSystem: %v", err)
+	}
+
+	ntpConfig := types.HostNtpConfig{
+		Server: servers,
+	}
+
+	dateTimeConfig := types.HostDateTimeConfig{
+		NtpConfig: &ntpConfig,
+	}
+
+	return s.UpdateConfig(ctx, dateTimeConfig)
+}
+
+func readHostNtpServerConfig(ctx context.Context, client *govmomi.Client, hostObject *object.HostSystem) ([]string, error) {
+	// Retrieve the host's configuration
+	var hostSystem mo.HostSystem
+	err := hostObject.Properties(ctx, hostObject.Reference(), []string{"config.dateTimeInfo"}, &hostSystem)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get host configuration: %v", err)
+	}
+
+	// Check if the dateTimeInfo configuration is available
+	if hostSystem.Config == nil || hostSystem.Config.DateTimeInfo == nil {
+		return nil, fmt.Errorf("dateTimeInfo configuration is not available on host")
+	}
+
+	// Check if the NTP configuration is available
+	if hostSystem.Config.DateTimeInfo.NtpConfig == nil {
+		return nil, fmt.Errorf("NTP configuration is not available for the host")
+	}
+
+	// Log the NTP servers found (optional)
+	fmt.Printf("NTP Servers for host: %v\n", hostSystem.Config.DateTimeInfo.NtpConfig.Server)
```

Comment:
I don't think fmt.Printf ends up in the TF logs.
I suggest using  log.XXX or even tflog.XXX

Reply:
i removed all the fmt.Printf code, noticed not needed now that it is functioning.